src/app/shared/background.model.ts

```diff
@@ -1,15 +1,44 @@
-import { Die, } from './die.model';
+import { Die } from './die.model';
 
 export class CharacterBackground {
-    constructor(
-    public number: number,
-    public name: string,
-    public backgroundText: string,
-    public qualities: string[],
-    public principleCategory: string,
-    public powerSourceDice: Die[],
-    public diceText: string,
-    public pageNumber: number,
-    public diceToAssign: Die[] = [],
-    ) {}
+    public number: number;
+    public name: string;
+    public backgroundText: string;
+    public qualities: string[];
+    public principleCategory: string;
+    public powerSourceDice: Die[] = [];
+    public diceText: string;
+    public pageNumber: number;
+    public diceToAssign: Die[] = [];
+
+    constructor(backgroundData: {
+        num: number;
+        name: string;
+        qualities: string[];
+        quality_dice: number[];
+        principle: string;
+        power_source_dice: number[];
+        page: number;
+        blurb: string;
+        background_text: string;
+        dice_text: string;
+    }) {
+        this.number = backgroundData.num;
+        this.name = backgroundData.name;
+        this.backgroundText = backgroundData.background_text;
+        this.qualities = backgroundData.qualities;
+        this.principleCategory = backgroundData.principle;
+        this.diceText = backgroundData.dice_text;
+        this.pageNumber = backgroundData.page;
+
+        for (let dieSides of backgroundData.power_source_dice) {
```

Comment:
there are some handy methods for working with arrays that make this simpler.  What you are doing in your code is looping through the array in `power_source_die` and changing each item to something else and creating a new array with those new items.  That can be done easily with [array.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)

replace
```
        for (let dieSides of backgroundData.power_source_dice) {
            let die = new Die(dieSides);
            this.powerSourceDice.push(die);
        }
```
with
```
        this.powerSourceDice = backgroundData.power_source_dice.map((dieSides) => new Die(dieSides));
```

Reply:
I knew this felt inefficient and I'll add this to the list of new skills to implement